### ISSUE TITLE
Let users specify their own message priority type instead of u32 (#85)

### ISSUE THREAD

**Issue by Restioson** [Restioson/xtra#92]

**Comment by Restioson** [comment]
> Why is u32 not good enough? Are you thinking of custom priorities with enums?

Yes, this is what I was thinking. It might be a bit easier to reason about

> u32 might even be a bit much for priority, perhaps u8 is enough?

That's true, but maybe it's better for now to err on the side of more rather than less?

**Comment by thomaseizinger** [comment]
An advantage of a fixed set of priorities would be that we can guarantee FIFO across a single priority because we can implement it by just having one queue per priority, i.e. _not_ mix messages of different priorities.

What do you think of that? I quite like the simplicity of having only "low", "default" and "high" priority messages.

**Comment by Restioson** [comment]
That's true. I mostly worry about the flexibility concerns here.

**Comment by thomaseizinger** [comment]
> That's true. I mostly worry about the flexibility concerns here. 

Yeah I can see that. Given how much it could simplify internally, what do you think of trimming the feature down initially and see if someone complains about the lack of more priorities?

Even if we start with low, default and high, we can always add 1-2 more before having to switch conceptually.
